server/controllers/finance/cash.create.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const uuid = require('node-uuid');
+const _ = require('lodash');
+const db   = require('../../lib/db');
+const BadRequest = require('../../lib/errors/BadRequest');
+const util = require('../../lib/util');
+
+// @fixme - only for testing purposes
+const q = require('q');
+
+module.exports = create;
+
+/**
+ * @function processCashItems
+ *
+ * @description
+ * This method prepares the cash items for writing to the DB.  It ensures that
+ * the uuids are defined and converted, and that each item represents an ordered
+ * array of values.
+ *
+ * @param {Buffer} cashUuid - the binary uuid of the cash record
+ * @param {Array} items - an array of cash_item JSONs
+ * @returns {Array} - an ordered array of arrays (cash_items)
+ */
+function processCashItems(cashUuid, items) {
+
+  // make sure uuids are defined and converted
+  items.forEach(item => {
+    item.cash_uuid = cashUuid;
+    item.uuid = item.uuid || uuid.v4();
+
+    item = db.convert(item, ['uuid', 'invoice_uuid']);
+  });
+
+  // make sure the items are in an ordered array
```

Comment:
It is completely necessary.  The values are expected to be inserted in a specific order ('uuid', 'cash_uuid', 'amount', 'invoice_uuid') and JSON is an unordered format.  [See this answer](http://stackoverflow.com/a/5525820).